pyproject.toml

```diff
@@ -21,6 +21,8 @@ pyspark = "^3.3.1"
 regex = "^2023.5.5"
 urllib3 = "<=2.0"
 sphinxcontrib-bibtex = "^2.5.0"
+zstandard = "^0.21.0"
+blake3 = "^0.3.3"
```

Comment:
Could you also explain why blake3 is needed? It may come from a performance standpoint. All I want is to ensure I have a correct understanding for future reference and documentation.

Reply:
My logic was 
md5 -> xxhash (insecure but fast)
sha1 -> blake3 (secure but less fast)
Also both xxhash and blake3 can be seeded to actually make `permutations` number of hashes instead of simulating it through permutations.

However, I realised that such permutations are theoretically almost as good as a new hash (as long as the permutation source hash is good) and much faster. Also hashing is NOT the bottleneck here. IT is python related string manipulations (not a lot we can do about here).
Also blake3 causes a LOT of problems with pickling (it's structure looks like this blake3.blake3.blake3... and pickle hates that).
I'm going to revert this one.